File: FileReverse-C/FileReverse.java
/**
 * FileReverse.java
 * Purpose:
   Reads lines of input and prints tokens backward to output file
 * Author: Some codes obtained from Lab 2 instructions file FileTokens.java
 * Zifei (Alice) Lu
   alicelu625@gmail.com
 * 10/5/2017
 */

import java.io.*;
import java.util.Scanner;

class FileReverse {
	public static void main(String[] args) throws IOException {
		
		int lineNumber = 0;
		
		//check number of command line arguments is at least 2
		if (args.length < 2) {
			System.out.println("Usage: java -jar FileReverse.jar <input file> <output file>");
			System.exit(1);
		}
		
		//open files
		Scanner in = new Scanner(new File(args[0]));
		PrintWriter out = new PrintWriter(new FileWriter(args[1]));
		
		//read lines from in, extract & print tokens from each line
		while (in.hasNextLine()) {
			lineNumber++;
			//trim leading & trailing spaces, then add one trailing space so split works on blank lines
			String line = in.nextLine().trim() + " ";
			//split line around white spaces
			String[] token = line.split("\\s+");
			//print out tokens
			int n = token.length;
			for (int i = 0; i < n; i++) {
				out.println(stringReverse(token[i]));
				//out.println(token[i]);
			}
		}
		
		//close files
		in.close();
		out.close();
	}
	
	public static String stringReverse(String s) {
		char[] line = s.toCharArray();
		char temp;
		int start, end = line.length-1;
		for (start = 0; start < end; start++) {
			temp = line[start];
			line[start] = line[end];
			line[end] = temp;
			end--;
		}
		return new String(line);
	}
}

File: FileReverse-C/fileReverse.c
/**
 * fileReverse.c
 * Purpose:
   Reads lines of input and prints tokens backward to output file
 * Zifei (Alice) Lu
   alicelu625@gmail.com
 * 10/21/2017
 */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void stringReverse(char* s);

int main(int argc, char* argv[]) {
	FILE* in; //file handle for input
	FILE* out; //file handle for output
	char word[256]; //char array to store words from input file

	//check command line for correct number of arguments
	if (argc != 3) {
		printf("Usage: %s <input file> <output file> \n", argv[0]);
		exit(EXIT_FAILURE);
	}

	//open input file for reading
	in = fopen(argv[1], "r");
	if (in == NULL) {
		printf("Unable to read from file %s \n", argv[1]);
		exit(EXIT_FAILURE);
	}

	//open output file for writing
	out = fopen(argv[2], "w");
	if (out == NULL) {
		printf("Unable to write to file %s \n", argv[2]);
		exit(EXIT_FAILURE);
	}

	//read words from input file, print on separate lines reversed to outputfile
	while (fscanf(in, " %s", word) != EOF) {
		stringReverse(word);
		fprintf(out, "%s\n", word);
	}

	//close input & output files
	fclose(in);
	fclose(out);
}


//reverses word
void stringReverse(char *s) {
	char temp;
	int start, end = strlen(s)-1;
	for (start = 0; start < end; start++) {
		temp = s[start];
		s[start] = s[end];
		s[end] = temp;
		end--;
	}
	return;
}